extension/llm/runner/text_llm_runner.h

```diff
@@ -0,0 +1,183 @@
+/*
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+// A simple llama2 runner that includes preprocessing and post processing logic.
+// The module takes in a string as input and emits a string as output.
+
+#pragma once
+
+#include <cstdint>
+#include <functional>
+#include <memory>
+#include <string>
+#include <unordered_map>
+
+#include <executorch/extension/llm/runner/irunner.h>
+#include <executorch/extension/llm/runner/stats.h>
+#include <executorch/extension/llm/runner/text_decoder_runner.h>
+#include <executorch/extension/llm/runner/text_prefiller.h>
+#include <executorch/extension/llm/runner/text_token_generator.h>
+#include <executorch/extension/module/module.h>
+#include <pytorch/tokenizers/tokenizer.h>
+
+namespace executorch::extension::llm {
+
+class ET_EXPERIMENTAL TextLLMRunner : public IRunner {
+ public:
+  /**
+   * @brief Constructor for TextLLMRunner with dependency injection
+   *
+   * Creates a TextLLMRunner instance with all required components for text
+   * generation.
+   *
+   * @param metadata Key-value pairs containing model metadata (e.g.,
+   * vocab_size, context_length)
+   * @param tokenizer Tokenizer for converting between text and token IDs
+   * @param module The underlying model module that performs inference
+   * @param text_decoder_runner Component responsible for running the decoder
+   * part of the model
+   * @param text_prefiller Component for handling the prefill phase of text
+   * generation
+   * @param text_token_generator Component for generating tokens during the
+   * decode phase
+   * @param stats Statistics tracking object for performance monitoring
+   * @param temperature Temperature parameter for controlling randomness in
+   * generation (deprecated). Please use GenerationConfig.temperature instead.
+   */
+  explicit TextLLMRunner(
+      std::unordered_map<std::string, int64_t> metadata,
+      std::unique_ptr<::tokenizers::Tokenizer> tokenizer,
+      std::unique_ptr<::executorch::extension::Module> module,
+      std::unique_ptr<TextDecoderRunner> text_decoder_runner,
+      std::unique_ptr<TextPrefiller> text_prefiller,
+      std::unique_ptr<TextTokenGenerator> text_token_generator,
+      std::unique_ptr<Stats> stats,
+      float temperature = -1.0f);
```

Comment:
if it's deprecated, why include this as part of the API?

If there's a user code, they can always just use create_text_llm_runner, or can we migrate the user code to use GenerationConfig directly?